internal/driver/glfw/loop.go

```diff
@@ -115,87 +116,98 @@ func (d *gLDriver) runGL() {
 		d.trayStart()
 	}
 	fyne.CurrentApp().Lifecycle().(*app.Lifecycle).TriggerStarted()
-	eventTick := time.NewTicker(time.Second / 60)
-	for {
-		select {
-		case <-d.done:
-			eventTick.Stop()
-			d.drawDone <- struct{}{} // wait for draw thread to stop
-			d.Terminate()
+
+	postEmptyEvent()
+
+	eventTick := time.NewTicker(time.Second / 60) // TODO: Why does it not work without this?
```

Comment:
Are you saying it works with the ticker but not the bare for{}?
If so my guess would be that its blocking the posting of other events that you're expecting will wake it up.
You could test by switching to `[glfwWaitEventsTimeout](https://www.glfw.org/docs/3.3/group__window.html#ga605a178db92f1a7f1a925563ef3ea2cf)(0.7);` and see if it helps

Reply:
> Are you saying it works with the ticker but not the bare for{}?

Yup. I am worried that `glfwWaitEventsTImout` will be just as bad as the previous code unless we have a very long timeout.

Reply:
It was not meant as a solution, merely more data for debugging

Reply:
I think postEmtpyEvent at line 120 needs to be in a goroutine so that we have called waitEvents before postEmptyEvent is called. Doing that seems to launch the window in my testing without using the ticker.

The real challenge is figuring out everywhere we would now need to PostEmtpyEvent to wake up the main thread. Especially on ARM Mac where we draw on the main thread it will be a challenge, since the waitEvents call will need to be woken up whenever there's something that needs to be drawn/refreshed.

Reply:
Ah, cool. I'll keep that in mind and look at this once again going forward